### ISSUE TITLE
Adding Terrain World and Terrain World Renderer components to a level causes an assert to be printed

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#15417]
**Describe the bug**
Adding *Terrain World* and *Terrain World Renderer* components to a level causes an assert to be printed to the Console.

Assert information:
File:
```
D:/workspace/o3de/Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialTypeAsset.cpp
```
Condition:
```
(3748) 'bool __cdecl AZ::RPI::MaterialTypeAsset::PostLoadInit(void)'
Failed to InitializeNonSerializedData
```
Line: 235

This issue does not occur on the Development (2c16b8c) build.

**Prerequisites**
* Make sure that the project has the Terrain Gem enabled.

**Steps to reproduce**
1. Create a new level.
2. Select the level in the Entity Outliner.
3. Add *Terrain World* and *Terrain World Renderer* components to it in the *Entity Inspector*.

**Expected behavior**
No asserts are printed to the Console.

**Actual behavior**
An assert is printed to the Console.

**Video**

https://user-images.githubusercontent.com/86952082/228240625-f3a6e377-d853-4f01-bf9a-a34202b39a1e.mp4

**Found in Branch**
Stabilization/2305 (5989990)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
5989990d8cbaea16849ed6386f52c47a0d2d663c

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 11
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by hultonha** [comment]
Discussed in sig-simulation triage. This looks more like a rendering issue to moving to @o3de/sig-graphics-audio to triage. Thanks!

**Comment by invertednormal** [comment]
@gadams3

**Comment by gadams3** [comment]
@galibzon I believe this relates to your recent changes

**Comment by galibzon** [comment]
PR Fix: https://github.com/o3de/o3de/pull/15499

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2305 (c4dd0ce) build.
